include/api/CDataFrameAnalysisInstrumentation.h

```diff
@@ -32,8 +32,6 @@ class API_EXPORT CDataFrameAnalysisInstrumentation
 public:
     CDataFrameAnalysisInstrumentation();
 
-    virtual ~CDataFrameAnalysisInstrumentation() = default;
```

Comment:
We have a pure virtual method in the `protected` area. That is why this virtual destructor is here.

Reply:
You only need to define it a virtual once in the base class (unless you need it to do something non-default). So the necessary cleanup is already ensured by `CDataFrameAnalysisInstrumentation`. This was generating a warning for me, I think because it wasn't marked as override, but I think removing it is a better fix.

Reply:
If you feel strongly I can add back in and mark as override, but this seems cleaner to me. WDYT?

Reply:
There's a whole discussion about this in https://stackoverflow.com/questions/40999791/default-override-of-virtual-destructor

In a large codebase the `static_assert` suggested might make sense, but I am not convinced it's worth it in our small codebase where the people changing the code generally know the class hierarchy in the bit they're changing very well.

One more data point is that in the days before `default` and `override` existed we wouldn't have declared a constructor in this situation.

Reply:
Alright :+1: